src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py

```diff
@@ -144,6 +144,21 @@ def load_arguments(self, _):
     with self.argument_context('group deployment operation show') as c:
         c.argument('operation_ids', nargs='+', help='A list of operation ids to show')
 
+    with self.argument_context('deployment') as c:
+        c.argument('deployment_name', options_list=('--name', '-n'), required=True, help='The deployment name.')
+        c.argument('deployment_location', options_list=('--location', '-l'), required=True, help='The deployment location.')
```

Comment:
options_list should use list syntax, not tuple.

Reply:
Not quite understand what the issue is.. can you give more details? I'm following what we do for "group deployment".

---
In reply to: [195173832](https://github.com/Azure/azure-cli/pull/6572#discussion_r195173832) [](ancestors = 195173832)

Reply:
It should look like `options_list=['--location', '-l']`

Reply:
Also, here's the metadata from the global location alias:
```Python
    location_type = CLIArgumentType(
        options_list=('--location', '-l'),
        completer=get_location_completion_list,
        type=get_location_name_type(cli_ctx),
        help="Location. You can configure the default location using `az configure --defaults location=<location>`",
        metavar='LOCATION',
        configured_default='location')
```

You should register your param like this:
```Python
c.argument('deployment_location', arg_type=get_location_type(self.cli_ctx), required=True)
```

If your command signature does not show a default value for location, you don't even need the `required=True` part. The CLI will automatically know the location is required.

Reply:
thanks. I've updated this one.